docs/service-and-repository-structure/service-and-repository-structure.md

```diff
@@ -0,0 +1,137 @@
+See [README](https://github.com/planetarium/NineChronicles.Headless/blob/main/README.md) for all available options.
+
+#### 3.2.1.2. Usage of MagicOnion
+To make gRPC connection between Unity client and .NET server, we chose [MagicOnion](https://cysharp.github.io/MagicOnion/) as an implementation.  
+This is well-made server to meet our requirements, and distributed under [MIT license](https://opensource.org/licenses/MIT).  
+Although we're not using MagicOnion's feature very well inside logic, but this information we're using MagicOnion is useful for first-time viewer.
+
+## 3.3. NineChronicles
+
+#### [GitHub repository](https://github.com/planetarium/NineChronicles)
+
+The unity client to play NineChronicles Game. You need `Unity` to build and run this game on your local computer.  
+Following [Getting Started](https://github.com/planetarium/NineChronicles/wiki/Get-Started), you can build and run your own NineChronicles client on your environment.  
+If you want to enhance UX/UI or any user interactive properties, this is the right repository to dig.
+
+### 3.3.1. Remark
+
+#### 3.3.1.1. Unity version issue
+
+Due to prevent unintended issue, we highly recommend to use Unity version as we wrote in the document.  
+If you use later version instead noted version, you can meet a bunch of errors, and we cannot help at that situation.
+
+#### 3.3.1.2. Select blockchain to attach
+
+If you run your game using `Play` button on your Unity editor, It will run with its own internal blockchain network, not any existing external blockchain.  
+To attach your client to mainnet or other network, you have to change client configurations.
+
+## 3.4. Lib9c
+
+#### [GitHub repository](https://github.com/planetarium/lib9c)
+
+This is common library used broadly among NineChronicles game server and client.  
+This is [`Libplanet`](#3.5.-Libplanet) implementation to run NineChronicles game on libplanet blockchain network.  
+In this library, you can see NineChronicles' status & action definitions and all game logics.  
+If you want to mod game action and/or logic, this repository is the one.
+
+## 3.5. Libplanet
+
+#### [GitHub repository](https://github.com/planetarium/libplanet)
+
+This is the very core library that is the blockchain protocol. Libplanet is written in C# and you can install it using [NuGet](https://www.nuget.org/packages/Libplanet/) package.  
+Everything to save and update is called `state`, and any type of state mutation is called `action`.  
+You can see the library design, tools and all API references in [Libplanet Documentation](https://docs.libplanet.io/).
+If you want to see the network structure, consensus algorithm and so on, you should dig this repository to meet what you seek.
+
+### 3.5.1. Remark
+
+#### 3.5.1.1. Difference between Libplanet and Lib9c
+
+- `Liblpanet` is the blockchain protocol and interface to run game on blockchain network.
+- `Lib9c` is the NineChronicles version implementation of `Libplanet`.
```

Comment:
Lib9c isn't a reimplementation of Libplanet but a library containing most of the game logic. You could think of the headless as a wrapper around the Lib9c to provide interfaces such as CLI, GraphQL and MagicOnion.

Reply:
I thought that lib9c has implementation of `IAction` and sush things so I used the word `implementation`. So you mean lib9c is not a 9c specific implementation at all?
I think I'm still confused about this difference.
@BasixKOR Can you suggest better wording for this section? :pray:

Reply:
I'll use the README of both repositories at this moment.